msg/templates/urtps/Publisher.cpp.em

```diff
@@ -127,17 +127,24 @@ bool @(topic)_Publisher::init()
 
 void @(topic)_Publisher::PubListener::onPublicationMatched(Publisher* pub, MatchingInfo& info)
 {
-    (void)pub;
-
-    if (info.status == MATCHED_MATCHING)
-    {
-        n_matched++;
-        std::cout << " - @(topic) publisher matched" << std::endl;
+    // The first 6 values of the ID guidPrefix of an entity in a DDS-RTPS Domain
+    // are the same for all its subcomponents (publishers, subscribers)
+    std::stringstream own_endpoint, remote_endpoint;
+    for (size_t i = 0; i < 6; i++) {
+        own_endpoint << pub->getGuid().guidPrefix.value[i];
```

Comment:
Rather than making two strings, just do the checks on the characters directly as you go, updating a bool. Allocating the `std::stringstream`s is pretty expensive compared to the rest of everything else here.

Reply:
Yes that makes sense

Reply:
@jkflying done https://github.com/PX4/Firmware/pull/14297/commits/da722324c5e480932681e78bc6d1517c31d5fac5